### PR TITLE
3.x into Master

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Right side of `while ($admin->isChild() && $admin = $admin->getParent());` is always true @phansys

Any idea about how I should change the do/while ?

**Comment by phansys** [comment]
>Right side of while ($admin->isChild() && $admin = $admin->getParent()); is always true @phansys

Yes, it is there just for the assignment.

>Any idea about how I should change the do/while ?

ATM, no. I could try to find other way later if it is required.

**Comment by VincentLanglet** [comment]
> > Any idea about how I should change the do/while ?
> 
> ATM, no. I could try to find other way later if it is required.

It's required indeed  since it's reported by phpstan ^^'

**Review by dmaicher — APPROVED** [review]
phpstan should be fine after another merge with 3.x?

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -1327,6 +1317,27 @@ protected function handleXmlHttpRequestSuccessResponse(Request $request, object
...
     }
 
+    final protected function assertObjectExists(Request $request): void
+    {
+        $admin = $this->admin;
+
+        do {
+            $objectId = $request->get($admin->getIdParameter());
+
+            if (null !== $objectId) {
```
Comment (wbloszyk) [inline src/Controller/CRUDController.php:1327]: Im not sure but IMO `null === $objectId` should throw exception.

**Review by dmaicher — DISMISSED** [review]

**Review by dmaicher — APPROVED** [review]

**Pull request merged**